libqaul/src/users/mod.rs

```diff
@@ -4,14 +4,35 @@ use identity::Identity;
 use rand::prelude::*;
 use std::collections::BTreeMap;
 
-/// User context
-pub(crate) struct User {
-    pub(crate) id: Identity,
-    pub(crate) display_name: Option<String>,
-    pub(crate) real_name: Option<String>,
-    pub(crate) bio: BTreeMap<String, String>,
-    pub(crate) services: Vec<String>,
-    pub(crate) avatar: Option<Vec<u8>>,
+/// A complete user, with ID and data.
+#[derive(Debug, PartialEq, Clone)]
+pub struct User {
+    /// A users network ID
+    pub id: Identity,
+    /// The user's information, such as name, bio, etc.
+    pub data: UserData,
```

Comment:
This is an interesting change. Initially I wanted to split the representation of a user `libqaul` internally and service API externally because internally we will probably also end up storing some secret stuff that we definitely don't want to accidentally leak outwards.

On the other had, we could probably just keep that state elsewhere and refer to it via the user id or something.

All in all, this feels more elegant, so :+1: